var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -48,67 +51,198 @@ class Hypre(Package):
+        configure_args.append(
+            '--with%s-fei' % ('' if '+fei' in spec else 'out'))
+        configure_args.append(
+            '--with%s-mli' % ('' if '+mli' in spec else 'out'))
+
+        configure_args.append(
+            '--with%s-openmp' % ('' if '+openmp' in spec else 'out'))
+        configure_args.append(
+            '--%s-complex' % ('enable' if '+complex' in spec else 'disable'))
+
+        if '@2.13.0+cuda' in spec:
+            # TODO: Use the 'cuda_arch' variant value.
+            # The required minimum is sm_30.
+            cuda_arch_flags = ['-arch', 'sm_30']
+            configure_args += ['--with-cuda', '--enable-unified-memory',
+                               '--with-nvcc']
+            # The nvcc compiler will be used for compiling all files
+            nvcc = str(spec['cuda'].prefix.bin.nvcc)
+            # Flags used for compiling mixed CUDA and C/C++ files with file
+            # extensions different from .cu
+            cflags_orig = list(cflags)
+            cflags = ['-O2', '--x cu', '-ccbin=%s' % env['CXX'],
+                      '--std=c++11', '-expt-extended-lambda',
+                      '-DUSE_NVTX', '-DHYPRE_USE_GPU', '-DHYPRE_USE_MANAGED',
+                      '-DHYPRE_USE_CUDA', '-DHAVE_CONFIG_H',
+                      '--relocatable-device-code=true']
+            if 'essl' in spec:
+                cflags += ['-D_ESVCPTR']
+            # Flags used for compiling .cu files
+            cuflags = ['-O2', '-ccbin=%s' % env['CXX'],
+                       '-I.', '-I..', '-I../utilities',
+                       '-I%s' % spec['mpi'].prefix.include,
+                       '-DUSE_NVTX', '-DHYPRE_USE_GPU',
+                       '-DHYPRE_USE_MANAGED']
+            # Flags common for cflags and cuflags
+            common_flags = list(cuda_arch_flags)
+            if '+shared' in spec:
+                common_flags += ['-Xcompiler', self.compiler.pic_flag]
+            if '+openmp' in spec:
+                common_flags += ['-Xcompiler', self.compiler.openmp_flag]
+            cflags += common_flags
+            cuflags += common_flags
+
+            cflags_joined = ' '.join(cflags + cflags_orig)
+            cxxflags_joined = ' '.join(cflags + cxxflags)
+            make_args += [
+                'CC=%s' % nvcc,
+                'CXX=%s' % nvcc,
+                'NVCC=%s' % nvcc,
+                'CFLAGS=%s $(C_COMPILE_FLAGS)' % cflags_joined,
+                'CXXFLAGS=%s $(CXX_COMPILE_FLAGS)' % cxxflags_joined,
+                'NVCCFLAGS=%s' % ' '.join(cuflags)]
+
+            # Set LDFLAGS - used for linking the shared library and for linking
+            # the tests.
+            cuda_lib_names = ['cusparse', 'cudart', 'cublas', 'nvToolsExt']
+            cuda_lib_dirs = [join_path(spec['cuda'].prefix, p)
+                             for p in ['lib64', 'lib']]
+            for dir in cuda_lib_dirs:
+                cuda_libs = find_libraries(
```

Comment:
those things should co to `cuda` package `libs()`

Reply:
I agree. Should I do it in this branch or a separate one?

Reply:
IMO this branch is ok